sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradesMatrixUnit.scala

```diff
@@ -129,11 +129,12 @@ abstract class UpgradesMatrixUnit(n: Int, k: Int)
     }
     val lookupContractByKey = contractOrigin match {
       case UpgradesMatrixCases.Global | UpgradesMatrixCases.Disclosed =>
-        val keyMap = Map(
+        val keyMap =
           testHelper
             .globalContractKeyWithMaintainers(setupData)
-            .globalKey -> setupData.globalContractId
-        )
+            .toList
+            .map(kwm => kwm.globalKey -> setupData.globalContractId)
+            .toMap
         ((kwm: GlobalKeyWithMaintainers) => keyMap.get(kwm.globalKey)).unlift
```

Comment:
This seems like it could be replaced with a flatMap that avoids generating a singleton map:
```
((kwm: GlobalKeyWithMaintainers) =>
  testHelper.globalContractKeyWithMaintainers(setupData)
    .flatMap(helperKey =>
      Option.when(helperKey.globalKey == kwm.globalKey)(setupData.globalContractId)
    )).unlift
```
I leave it up to you, though - I'm not very experienced with Scala so I'm not sure which is more "idiomatic" Scala.